components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/DefaultLoginValidationDelegate.kt

```diff
@@ -22,14 +23,20 @@ class DefaultLoginValidationDelegate(
     private val storage: Lazy<LoginsStorage>,
     private val scope: CoroutineScope = CoroutineScope(IO)
 ) : LoginValidationDelegate {
+    private val logger = Logger("DefaultLoginValidationDelegate")
 
     /**
      * Compares a [Login] to a passed in list of potential dupes [Login] or queries underlying
      * storage for potential dupes list of [Login] to determine if it should be updated or created.
      */
     override fun shouldUpdateOrCreateAsync(entry: LoginEntry): Deferred<Result> {
         return scope.async {
-            val foundLogin = storage.value.findLoginToUpdate(entry)
+            val foundLogin = try {
+                storage.value.findLoginToUpdate(entry)
+            } catch (e: LoginsStorageException) {
+                logger.error("Failure in shouldUpdateOrCreateAsync: $e")
```

Comment:
What kind of problems are you seeing here i.e., are you interested in logging these to Sentry or is there nothing we can do about them anyway. Then just logging here is fine!

Reply:
Without the try..catch, if Fenix ever sends an invalid login entry it crashes the app.  I definitely think logging to sentry makes sense in this case.  I thought `logger.error` would do that, what's the correct way?

Reply:
@bendk `logger.error` only logs to the Android system (android.util.log): https://github.com/mozilla-mobile/fenix/blob/f953c5ec94181f2af73688ce4d7e470fba5e3b62/app/src/main/java/org/mozilla/fenix/FenixApplication.kt#L173

If you want to submit this to Sentry, we need to pass in a crash reporter instance to `DefaultLoginValidationDelegate` here and call `submitCaughtException`. 

Here's an example:
https://github.com/mozilla-mobile/fenix/blob/f953c5ec94181f2af73688ce4d7e470fba5e3b62/app/src/main/java/org/mozilla/fenix/FenixApplication.kt#L173
https://github.com/mozilla-mobile/android-components/blob/main/components/feature/session/src/main/java/mozilla/components/feature/session/PictureInPictureFeature.kt#L55

This would need a quick PR to Fenix as well, but it's not breaking API as long as the `crashReporting` parameter is nullable and optional (null by default). This can be a follow-up as well. Up to you :).